src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -588,10 +588,14 @@ private SignatureData LoadSignature()
             }
 
             // Dynamify object type if necessary
-            paramInfo[0].Type = paramInfo[0].Type.AsDynamicIfNoPia(_containingType);
+            var returnType = paramInfo[0].Type = paramInfo[0].Type.AsDynamicIfNoPia(_containingType);
+
+            // Check for tuple type
+            returnType = TupleTypeDecoder.DecodeTupleTypesIfApplicable(returnType, paramInfo[0].Handle, moduleSymbol);
 
             var returnParam = PEParameterSymbol.Create(moduleSymbol, this, 0, paramInfo[0], out isBadParameter);
```

Comment:
In the case where we do have tuple names, does it matter that `paramInfo[0].Type` isn't updated?